ngraph/test/visitors/op/binary_elementwise.hpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <vector>
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+
+#include "ngraph/op/util/attr_types.hpp"
+#include "util/visitor.hpp"
+
+using namespace ngraph;
+using ngraph::test::NodeBuilder;
+template <typename T, element::Type_t ELEMENT_TYPE>
+class BinaryOperatorType
+{
+public:
+    using op_type = T;
+    static constexpr element::Type_t element_type = ELEMENT_TYPE;
+};
+template <typename T>
+class BinaryOperatorVisitor : public testing::Test
+{
+};
+
+class BinaryOperatorTypeName
+{
+public:
+    template <typename T>
+    static std::string GetName(int)
```

Comment:
On the first glance this function should be in `BinaryOperator` (`BinaryOperatorType`).

Reply:
It seems like we need a separate class for this function to pass it as an argument calling `INSTANTIATE_TYPED_TEST_SUITE_P`.

Reply:
Apparently yes.